cmd/storage-class.go

```diff
@@ -115,68 +128,39 @@ func parseStorageClass(storageClassEnv string) (sc storageClass, err error) {
 	return sc, nil
 }
 
-// Validates the parity disks for Reduced Redundancy storage class
-func validateRRSParity(rrsParity, ssParity int) (err error) {
-	disks := len(globalEndpoints)
-	// disks < 4 means this is not a erasure coded setup and so storage class is not supported
-	if disks < 4 {
+// Validates the parity disks.
+func validateParity(ssParity, rrsParity int) (err error) {
+	if ssParity == 0 && rrsParity == 0 {
+		return nil
+	}
+
+	if !globalIsXL && !globalIsDistXL {
 		return fmt.Errorf("Setting storage class only allowed for erasure coding mode")
 	}
 
-	// Reduced redundancy storage class is not supported for 4 disks erasure coded setup.
-	if disks == 4 && rrsParity != 0 {
-		return fmt.Errorf("Reduced redundancy storage class not supported for " + strconv.Itoa(disks) + " disk setup")
+	// SS parity disks should be greater than or equal to minimumParityDisks. Parity below minimumParityDisks is not recommended.
+	if ssParity < minimumParityDisks && ssParity > 0 {
```

Comment:
suggestion: `ssParity > 0 && ssParity < minimumParityDisks` reads more natural. In this form it says, if `ssParity` is not zero then check if it is lesser than `minimumParityDisks`.

Reply:
ok